tests/integration/extenders/PolicyTest.php

```diff
@@ -0,0 +1,174 @@
+                $this->normalUser(),
+            ],
+            'discussions' => [
+                ['id' => 1, 'title' => 'Unrelated Discussion', 'created_at' => Carbon::now()->toDateTimeString(), 'user_id' => 1, 'first_post_id' => 1, 'comment_count' => 1, 'is_private' => 0],
+            ],
+            'posts' => [
+                ['id' => 1, 'discussion_id' => 1, 'created_at' => Carbon::now()->toDateTimeString(), 'user_id' => 1, 'type' => 'comment', 'content' => '<t><p>a normal reply - too-obscure</p></t>'],
+            ]
+        ]);
+    }
+
+    /**
+     * @test
+     */
+    public function unrelated_user_cant_hide_discussion_by_default()
+    {
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('PATCH', '/api/discussions/1', $this->hideQuery)
+        );
+
+        $this->assertEquals(403, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function unrelated_user_can_hide_discussion_if_allowed()
+    {
+        $this->extend(
+            (new Extend\Policy(Discussion::class))
+                ->add(CustomPolicy::class)
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('PATCH', '/api/discussions/1', $this->hideQuery)
+        );
+
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function unrelated_user_cant_hide_discussion_if_denied()
+    {
+        $this->extend(
+            (new Extend\Policy(Discussion::class))
+                ->add(CustomPolicy::class)
+                ->add(DenyHidePolicy::class)
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('PATCH', '/api/discussions/1', $this->hideQuery)
+        );
+
+        $this->assertEquals(403, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function unrelated_user_can_hide_discussion_if_force_allowed()
+    {
+        $this->extend(
+            (new Extend\Policy(Discussion::class))
+                ->add(CustomPolicy::class)
+                ->add(DenyHidePolicy::class)
+                ->add(ForceAllowHidePolicy::class)
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('PATCH', '/api/discussions/1', $this->hideQuery)
+        );
+
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function unrelated_user_cant_hide_discussion_if_force_denied()
+    {
+        // Because the force deny policy is last, this also shows that
+        // order isn't considered, as the last result overrides all.
+        $this->extend(
+            (new Extend\Policy(Discussion::class))
+                ->add(CustomPolicy::class)
+                ->add(DenyHidePolicy::class)
+                ->add(ForceAllowHidePolicy::class)
+                ->add(ForceDenyHidePolicy::class)
```

Comment:
Should we have one test where they're scrambled? in all tests they seem to be registered in proper order (allow, deny, forceAllow, forceDeny).

Reply:
I agree that we should scramble them on purpose, just to make sure things get registered properly regardless of order, never trust people inputs.